packages/terminal/src/node/shell-terminal-server.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { ILogger } from '@theia/core/lib/common/logger';
+import { IShellTerminalServerOptions } from '../common/shell-terminal-protocol';
+import { BaseTerminalServer } from '../node/base-terminal-server';
+import { ShellProcessFactory } from '../node/shell-process'
+import { ProcessManager } from '@theia/core/lib/node/process/process-manager';
+
+@injectable()
+export class ShellTerminalServer extends BaseTerminalServer {
+
+    constructor(
+        @inject(ShellProcessFactory) protected readonly shellFactory: ShellProcessFactory,
+        @inject(ProcessManager) processManager: ProcessManager,
+        @inject(ILogger) logger: ILogger) {
+        super(processManager, logger);
+    }
+
+    create(options: IShellTerminalServerOptions): Promise<number> {
+        try {
+            const term = this.shellFactory(
```

Comment:
it seems like options can be passed down: `this.shellFactory(options)`

Reply:
yes I guess it could, I was not doing that because of rootURI... but I guess it doesn't hurt to pass a superset

Reply:
I think I'll make this more official by extending the options from shell-process and do the same thing for the terminal service

Reply:
Humm maybe not after all hehe this would complicate the extensions of the classes...

Reply:
not really critical, was just a suggestion